optuna/trial.py

```diff
@@ -351,6 +351,10 @@ def report(self, value, step=None):
         If step is set to :obj:`None`, the value is stored as a final value of the trial.
         Otherwise, it is saved as an intermediate value.
 
+        Note that the reported value is converted to ``float`` type by applying ``float()``
+        function internally. Thus, it accepts all float-like types (e.g., ``numpy.float32``).
```

Comment:
IMO, this example (``numpy.float32``) is not appropriate if the reported value contains only one element.

```python
>>> import numpy as np
>>> float(np.ndarray([1], np.float32))
1.0
>>> float(np.ndarray([1, 2], np.float32))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: only size-1 arrays can be converted to Python scalars
```

Reply:
Hmm, the type of `np.ndarray([1], np.float32)` is `np.ndarray` and I didn't think about it.
```python
>>> import numpy as np

# The type you mentioned.
>>> type(np.ndarray([1], np.float32))
<class 'numpy.ndarray'>

# The type I imagined.
>>> type(np.ndarray([1], np.float32)[0])
<class 'numpy.float32'>
```
Anyway, If the current phrase is confusing, I want to improve it.

Reply:
Oops! I have totally mistaken about the message. Some ML libraries return `np.ndarray` with a single `np.float32` element, and I automatically replaced `np.float32` with it when I read it.
Thank you for your kind explanation.